### PR TITLE
Auth-challenge-update

### PR DESCRIPTION
**Pull request by BrandonV21**

### CODE REVIEW

File: src/server/controllers/movie.js [hunk 1]
```diff
@@ -13,14 +13,34 @@ const getAllMovies = async (req, res) => {
 const createMovie = async (req, res) => {
     const { title, description, runtimeMins } = req.body;
 
+    const authorization = req.headers['authorization']
```
Comment (elvisonob) [inline src/server/controllers/movie.js:16]: Hi, what does this authorization variable do?

File: src/client/App.jsx [hunk 2]
```diff
@@ -15,15 +15,86 @@ function App() {
...
+        password: password
+      })
+    }
+
+    fetch(apiUrl + '/user/register', options)
+      .then(res => res.json())
+      .then(json => (console.log('Registered user:' + json.data.username)))
+      .catch((error) => {
+        //If the server is unavailable, then show an error to the user
+        setRegisterResponse('Server error')
```
Comment (Khateeb1111) [inline src/client/App.jsx:37]: Nice job in getting an error message to display. Good Work :)

File: src/client/App.jsx [hunk 3]
```diff
@@ -15,15 +15,86 @@ function App() {
...
   };
   
   const handleCreateMovie = async ({ title, description, runtimeMins }) => {
-    
+    const options = {
+      //Send a post request
+      method: 'POST',
+      //Set the headers
+      headers : {
+        'Authorization': 'Bearer' + localStorage.getItem('jwt')
```
Comment (Khateeb1111) [inline src/client/App.jsx:72]: What is this doing, not seen this before?

File: src/server/controllers/movie.js (hunk 1 above)
Comment (BrandonV21) [reply src/server/controllers/movie.js:16]: If you have a look at line 72, you can get the value of the requested authorised token.

File: src/server/controllers/user.js [hunk 4]
```diff
@@ -8,27 +8,39 @@ const jwtSecret = 'mysecret';
...
+
+    const createdUser = await prisma.user.create({
+        data: {
+            username: username,
+            password: hashedPassword
+        }
+    });
 
     res.json({ data: createdUser });
+    // res.json({ 'User created!': createdUser });
```
Comment (New-Arday) [inline src/server/controllers/user.js:21]: I liked your comments through out the file  but not sure what this commented code is useful

File: src/server/controllers/user.js [hunk 5]
```diff
@@ -8,27 +8,39 @@ const jwtSecret = 'mysecret';
...
+    const passwordsMatch = await bcrypt.compare(password, foundUser.password);    
 
     if (!passwordsMatch) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
 
-    const token = null;
+    const token = jwt.sign({ username }, jwtSecret);
 
     res.json({ data: token });
```
Comment (New-Arday) [inline src/server/controllers/user.js:45]: Clean and simple code to understand - good job!

**Pull request closed without merging**